### PR TITLE
[libc][math][c23] Add MPFR exhaustive test for fmodf16

### PR DESCRIPTION
**Pull request by overmighty**

### CODE REVIEW

File: libc/test/src/math/exhaustive/exhaustive_test.h [hunk 1]
```diff
@@ -68,6 +69,43 @@ struct UnaryOpChecker : public virtual LIBC_NAMESPACE::testing::Test {
...
+      StorageType ybits = y_start;
+      do {
+        FloatType y = FPBits(ybits).get_val();
+        mpfr::BinaryInput<FloatType> input{x, y};
+        bool correct = TEST_MPFR_MATCH_ROUNDING_SILENTLY(Op, input, FUNC(x, y),
+                                                         0.5, rounding);
+        failed += (!correct);
+        // Uncomment to print out failed values.
+        // if (!correct) {
+        //   TEST_MPFR_MATCH(Op::Operation, x, Op::func(x, y), 0.5, rounding);
```
Comment (overmighty) [inline libc/test/src/math/exhaustive/exhaustive_test.h:101]: TODO: Check that the commented code works.

File: libc/test/src/math/exhaustive/exhaustive_test.h [hunk 2]
```diff
@@ -167,6 +205,118 @@ struct LlvmLibcExhaustiveMathTest
...
 
+template <typename Checker>
+struct LlvmLibcBinaryInputExhaustiveMathTest
+    : public virtual LIBC_NAMESPACE::testing::Test,
+      public Checker {
+  using FloatType = typename Checker::FloatType;
+  using FPBits = typename Checker::FPBits;
+  using StorageType = typename Checker::StorageType;
+
+  static constexpr StorageType Increment = (1 << 2);
```
Comment (overmighty) [inline libc/test/src/math/exhaustive/exhaustive_test.h:216]: TODO: Maybe this should be a template parameter.

File: libc/test/src/math/exhaustive/exhaustive_test.h [hunk 3]
```diff
@@ -8,6 +8,7 @@
 
 #include "src/__support/CPP/type_traits.h"
 #include "src/__support/FPUtil/FPBits.h"
+#include "src/__support/macros/properties/types.h"
```
Comment (overmighty) [inline libc/test/src/math/exhaustive/exhaustive_test.h:11]: TODO: Add this to the `exhaustive_test` target dependencies in CMakeLists.txt.

**Comment by overmighty** [comment]
Rebased, resolved merge conflicts, and refactored the changes in exhaustive_test.h a lot.

**Comment by overmighty** [comment]
cc @lntue

File: libc/test/src/math/exhaustive/exhaustive_test.h [hunk 4]
```diff
@@ -68,22 +69,84 @@ struct UnaryOpChecker : public virtual LIBC_NAMESPACE::testing::Test {
...
+template <typename Checker, size_t Increment = 1 << 20>
 struct LlvmLibcExhaustiveMathTest
     : public virtual LIBC_NAMESPACE::testing::Test,
       public Checker {
   using FloatType = typename Checker::FloatType;
   using FPBits = typename Checker::FPBits;
   using StorageType = typename Checker::StorageType;
 
-  static constexpr StorageType INCREMENT = (1 << 20);
+  static constexpr StorageType INCREMENT = Increment;
```
Comment (overmighty) [inline libc/test/src/math/exhaustive/exhaustive_test.h:119]: Should probably get rid of this and use the template argument directly.

**Comment by overmighty** [comment]
Rebased.

**Review by lntue — APPROVED** [review]

**Pull request merged**